src/schemas/hierarchical-sets.schema.json

```diff
@@ -1,40 +1,71 @@
 {
-    "$schema": "http://json-schema.org/draft-07/schema#",
-    "$id": "https://github.com/hubmapconsortium/vitessce/#hierarchical-sets",
-    "title": "Vitessce hierarchical sets data",
-    "type": "object",
-    "definitions": {
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://github.com/hubmapconsortium/vitessce/#hierarchical-sets",
+  "title": "Vitessce hierarchical sets data",
+  "type": "object",
+  "definitions": {
       "stringArray": {
-        "type": "array",
-        "items": { "type": "string" }
+          "type": "array",
+          "items": { "type": "string" }
       },
       "colorArray": {
           "type": "array",
           "items": { "type": "integer", "minimum": 0, "maximum": 255 },
           "minItems": 3,
           "maxItems": 3
-      }
-    },
-    "additionalProperties": false,
-    "required": ["datasetId", "version", "setsType", "setsTree"],
-    "properties": {
-      "datasetId": { "type": "string" },
-      "version": { "type": "string" },
-      "setsType": { "type": "string" },
-      "setsTree": {
+      },
+      "treeLevelOnePlusNonLeaf": {
+          "type": "array",
+          "items": {
+              "type": "object",
+              "additionalProperties": false,
+              "required": ["name"],
+              "properties": {
+                  "name": { "type": "string" },
+                  "color": { "$ref": "#/definitions/colorArray" },
+                  "children": { "$ref": "#/definitions/treeLevelOnePlus" }
+              }
+          }
+      },
+      "treeLevelOnePlusLeaf": {
           "type": "array",
           "items": {
-            "type": "object",
-            "additionalProperties": false,
-            "required": ["key", "name"],
-            "properties": {
-              "key": { "type": "string" },
-              "name": { "type": "string" },
-              "color": { "$ref": "#/definitions/colorArray" },
-              "set": { "$ref": "#/definitions/stringArray" }
-            }
+              "type": "object",
+              "additionalProperties": false,
+              "required": ["name"],
+              "properties": {
+                  "name": { "type": "string" },
+                  "color": { "$ref": "#/definitions/colorArray" },
+                  "set": { "$ref": "#/definitions/stringArray" }
+              }
           }
-        }
-    }
+      },
+      "treeLevelOnePlus": {
+          "oneOf": [
+              { "$ref": "#/definitions/treeLevelOnePlusNonLeaf" },
+              { "$ref": "#/definitions/treeLevelOnePlusLeaf" }
+          ]
```

Comment:
It might be easier to treat this as a single type, and not worry so much about mutual exclusivity... Not sure.